compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -413,6 +413,14 @@ class TreePickler(pickler: TastyPickler) {
                 writeByte(if name.isTypeName then SELECTtpt else SELECT)
                 pickleNameAndSig(name, sig, ename)
                 pickleTree(qual)
+              else if defn.wasPolymorphicSignature(tree.symbol) then
+                writeByte(SELECTinPoly)
+                withLength {
+                  pickleNameAndSig(name, tree.symbol.signature, ename)
```

Comment:
Is `tree.symbol.signature` meaningful here? Isn't `tree.symbol` one of those automatically-created `Symbol`s, and hence `signature` would be the polymorphized one, from which we cannot find the symbol again?